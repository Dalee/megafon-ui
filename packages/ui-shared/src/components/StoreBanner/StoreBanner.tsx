import * as React from 'react';
import { cnCreate, Grid, GridColumn, Header, Paragraph, Link } from '@megafon/ui-core';
import './StoreBanner.less';
import appStoreImage from './img/app-store.png';
import googlePlayImage from './img/google-play.png';

export const Theme = {
    CLEAR_WHITE: 'clearWhite',
    GREEN: 'green',
    SPB_SKY_1: 'spbSky1',
} as const;

type ThemeType = typeof Theme[keyof typeof Theme];

export const DeviceMask = {
    ANDROID: 'android',
    NEW_IPHONE: 'new-iphone',
    BLACK_IPHONE: 'black-iphone',
    WHITE_IPHONE: 'white-iphone',
};

type DeviceMaskType = typeof DeviceMask[keyof typeof DeviceMask];

export interface IStoreBannerProps {
    /** Заголовок */
    title: string;
    /** Текст */
    text: string;
    /** Ссылка на App Store */
    linkApple: string;
    /** Ссылка на Google Play */
    linkGoogle: string;
    /** QR-код */
    qrCode?: string;
    /** Цветовая тема компонента */
    theme?: ThemeType;
    /** Изображение телефона */
    deviceMask: DeviceMaskType;
    /** Изображение на дисплее телефона */
    imageSrc: string;
}

const cn = cnCreate('mfui-beta-store-banner');
const StoreBanner: React.FC<IStoreBannerProps> = ({
    title,
    text,
    linkApple,
    linkGoogle,
    qrCode,
    imageSrc,
    theme = Theme.CLEAR_WHITE,
    deviceMask,
}) => (
    <div className={cn({ theme, mask: deviceMask })}>
        <div className={cn('container')}>
            <Grid className={cn('grid')}>
                <GridColumn
                        all="6"
                        mobile="12"
                        rightOffsetWide="1"
                        className={cn('content-wrapper')}
                >
                    <div className={cn('content')}>
                        <Header as="h2" className={cn('title')} color="inherit">
                            {title}
                        </Header>
                        <Paragraph className={cn('text')} hasMargin={false} color="inherit">
                            {text}
                        </Paragraph>
                        <div className={cn('links')}>
                            {qrCode && (
                                <img src={qrCode} className={cn('qr-code')} alt="QR-код"/>
                            )}
                            <div className={cn('stores')}>
                                <Link href={linkApple} className={cn('store-link', { 'app-store': true })}>
                                    <img src={appStoreImage} alt="Ссылка на приложение в App Store"/>
                                </Link>
                                <Link href={linkGoogle} className={cn('store-link')}>
                                    <img src={googlePlayImage} alt="Ссылка на приложение в Google Play" />
                                </Link>
                            </div>
                        </div>
                    </div>
                </GridColumn>
                <GridColumn all="4" desktop="6" tablet="6" mobile="12">
                    <div className={cn('device-wrapper')}>
                        <div className={cn('device')}>
                            <img
                                src={imageSrc}
                                className={cn('screen')}
                                alt="Изображение приложения на экране телефона"
                            />
                        </div>
                    </div>
                </GridColumn>
            </Grid>
        </div>
    </div>
);

export default StoreBanner;
