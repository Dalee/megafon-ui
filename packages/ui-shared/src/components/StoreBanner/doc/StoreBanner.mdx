---
name: StoreBanner
menu: Конструктор
route: components/store-banner
---

import Example from './StoreBanner.example.mdx';
import Props from './StoreBanner.props.mdx';

# BenefitsPictures

<CopyToClipboardBox text="import { StoreBanner } from '@megafon/ui-shared';" />

<DoczTabs>
    <Example />
    <Props />
</DoczTabs>
