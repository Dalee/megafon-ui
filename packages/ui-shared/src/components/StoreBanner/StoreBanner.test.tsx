import * as React from 'react';
import { shallow } from 'enzyme';
import StoreBanner, { IStoreBannerProps, DeviceMask, Theme } from './StoreBanner';

const props: IStoreBannerProps = {
    title: 'Title',
    text: 'Text',
    linkApple: 'App Store link',
    linkGoogle: 'Google Play link',
    deviceMask: DeviceMask.ANDROID,
    imageSrc: 'image.png',
};

describe('StoreBanner', () => {
    it('should render with required props', () => {
        const wrapper = shallow(<StoreBanner {...props} />);

        expect(wrapper).toMatchSnapshot();
    });

    it('should render with iphone mask', () => {
        const currentProps = {
            ...props,
            deviceMask: DeviceMask.BLACK_IPHONE,
        };
        const wrapper = shallow(<StoreBanner {...currentProps} />);

        expect(wrapper).toMatchSnapshot();
    });

    it('should render with green theme', () => {
        const currentProps = {
            ...props,
            theme: Theme.GREEN,
        };
        const wrapper = shallow(<StoreBanner {...currentProps} />);

        expect(wrapper).toMatchSnapshot();
    });

    it('should render with QR-code', () => {
        const wrapper = shallow(<StoreBanner {...props} qrCode={'qr-code.png'}/>);

        expect(wrapper).toMatchSnapshot();
    });
});
