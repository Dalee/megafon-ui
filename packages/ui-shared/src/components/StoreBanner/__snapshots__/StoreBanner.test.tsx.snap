// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`StoreBanner should render with QR-code 1`] = `
<div
  className="mfui-beta-store-banner mfui-beta-store-banner_theme_clearWhite mfui-beta-store-banner_mask_android"
>
  <div
    className="mfui-beta-store-banner__container"
  >
    <Grid
      className="mfui-beta-store-banner__grid"
      multiRow={true}
    >
      <GridColumn
        all="6"
        className="mfui-beta-store-banner__content-wrapper"
        flex={false}
        grow={false}
        mobile="12"
        orderAll="0"
        orderDesktop="0"
        orderMobile="0"
        orderTablet="0"
        orderWide="0"
        rightOffsetWide="1"
      >
        <div
          className="mfui-beta-store-banner__content"
        >
          <Header
            as="h2"
            className="mfui-beta-store-banner__title"
            color="inherit"
            hAlign="inherit"
          >
            Title
          </Header>
          <Paragraph
            className="mfui-beta-store-banner__text"
            color="inherit"
            hasMargin={false}
          >
            Text
          </Paragraph>
          <div
            className="mfui-beta-store-banner__links"
          >
            <img
              alt="QR-код"
              className="mfui-beta-store-banner__qr-code"
              src="qr-code.png"
            />
            <div
              className="mfui-beta-store-banner__stores"
            >
              <Link
                className="mfui-beta-store-banner__store-link mfui-beta-store-banner__store-link_app-store"
                href="App Store link"
              >
                <img
                  alt="Ссылка на приложение в App Store"
                  src="test-file-stub"
                />
              </Link>
              <Link
                className="mfui-beta-store-banner__store-link"
                href="Google Play link"
              >
                <img
                  alt="Ссылка на приложение в Google Play"
                  src="test-file-stub"
                />
              </Link>
            </div>
          </div>
        </div>
      </GridColumn>
      <GridColumn
        all="4"
        desktop="6"
        flex={false}
        grow={false}
        mobile="12"
        orderAll="0"
        orderDesktop="0"
        orderMobile="0"
        orderTablet="0"
        orderWide="0"
        tablet="6"
      >
        <div
          className="mfui-beta-store-banner__device-wrapper"
        >
          <div
            className="mfui-beta-store-banner__device"
          >
            <img
              alt="Изображение приложения на экране телефона"
              className="mfui-beta-store-banner__screen"
              src="image.png"
            />
          </div>
        </div>
      </GridColumn>
    </Grid>
  </div>
</div>
`;

exports[`StoreBanner should render with green theme 1`] = `
<div
  className="mfui-beta-store-banner mfui-beta-store-banner_theme_green mfui-beta-store-banner_mask_android"
>
  <div
    className="mfui-beta-store-banner__container"
  >
    <Grid
      className="mfui-beta-store-banner__grid"
      multiRow={true}
    >
      <GridColumn
        all="6"
        className="mfui-beta-store-banner__content-wrapper"
        flex={false}
        grow={false}
        mobile="12"
        orderAll="0"
        orderDesktop="0"
        orderMobile="0"
        orderTablet="0"
        orderWide="0"
        rightOffsetWide="1"
      >
        <div
          className="mfui-beta-store-banner__content"
        >
          <Header
            as="h2"
            className="mfui-beta-store-banner__title"
            color="inherit"
            hAlign="inherit"
          >
            Title
          </Header>
          <Paragraph
            className="mfui-beta-store-banner__text"
            color="inherit"
            hasMargin={false}
          >
            Text
          </Paragraph>
          <div
            className="mfui-beta-store-banner__links"
          >
            <div
              className="mfui-beta-store-banner__stores"
            >
              <Link
                className="mfui-beta-store-banner__store-link mfui-beta-store-banner__store-link_app-store"
                href="App Store link"
              >
                <img
                  alt="Ссылка на приложение в App Store"
                  src="test-file-stub"
                />
              </Link>
              <Link
                className="mfui-beta-store-banner__store-link"
                href="Google Play link"
              >
                <img
                  alt="Ссылка на приложение в Google Play"
                  src="test-file-stub"
                />
              </Link>
            </div>
          </div>
        </div>
      </GridColumn>
      <GridColumn
        all="4"
        desktop="6"
        flex={false}
        grow={false}
        mobile="12"
        orderAll="0"
        orderDesktop="0"
        orderMobile="0"
        orderTablet="0"
        orderWide="0"
        tablet="6"
      >
        <div
          className="mfui-beta-store-banner__device-wrapper"
        >
          <div
            className="mfui-beta-store-banner__device"
          >
            <img
              alt="Изображение приложения на экране телефона"
              className="mfui-beta-store-banner__screen"
              src="image.png"
            />
          </div>
        </div>
      </GridColumn>
    </Grid>
  </div>
</div>
`;

exports[`StoreBanner should render with iphone mask 1`] = `
<div
  className="mfui-beta-store-banner mfui-beta-store-banner_theme_clearWhite mfui-beta-store-banner_mask_black-iphone"
>
  <div
    className="mfui-beta-store-banner__container"
  >
    <Grid
      className="mfui-beta-store-banner__grid"
      multiRow={true}
    >
      <GridColumn
        all="6"
        className="mfui-beta-store-banner__content-wrapper"
        flex={false}
        grow={false}
        mobile="12"
        orderAll="0"
        orderDesktop="0"
        orderMobile="0"
        orderTablet="0"
        orderWide="0"
        rightOffsetWide="1"
      >
        <div
          className="mfui-beta-store-banner__content"
        >
          <Header
            as="h2"
            className="mfui-beta-store-banner__title"
            color="inherit"
            hAlign="inherit"
          >
            Title
          </Header>
          <Paragraph
            className="mfui-beta-store-banner__text"
            color="inherit"
            hasMargin={false}
          >
            Text
          </Paragraph>
          <div
            className="mfui-beta-store-banner__links"
          >
            <div
              className="mfui-beta-store-banner__stores"
            >
              <Link
                className="mfui-beta-store-banner__store-link mfui-beta-store-banner__store-link_app-store"
                href="App Store link"
              >
                <img
                  alt="Ссылка на приложение в App Store"
                  src="test-file-stub"
                />
              </Link>
              <Link
                className="mfui-beta-store-banner__store-link"
                href="Google Play link"
              >
                <img
                  alt="Ссылка на приложение в Google Play"
                  src="test-file-stub"
                />
              </Link>
            </div>
          </div>
        </div>
      </GridColumn>
      <GridColumn
        all="4"
        desktop="6"
        flex={false}
        grow={false}
        mobile="12"
        orderAll="0"
        orderDesktop="0"
        orderMobile="0"
        orderTablet="0"
        orderWide="0"
        tablet="6"
      >
        <div
          className="mfui-beta-store-banner__device-wrapper"
        >
          <div
            className="mfui-beta-store-banner__device"
          >
            <img
              alt="Изображение приложения на экране телефона"
              className="mfui-beta-store-banner__screen"
              src="image.png"
            />
          </div>
        </div>
      </GridColumn>
    </Grid>
  </div>
</div>
`;

exports[`StoreBanner should render with required props 1`] = `
<div
  className="mfui-beta-store-banner mfui-beta-store-banner_theme_clearWhite mfui-beta-store-banner_mask_android"
>
  <div
    className="mfui-beta-store-banner__container"
  >
    <Grid
      className="mfui-beta-store-banner__grid"
      multiRow={true}
    >
      <GridColumn
        all="6"
        className="mfui-beta-store-banner__content-wrapper"
        flex={false}
        grow={false}
        mobile="12"
        orderAll="0"
        orderDesktop="0"
        orderMobile="0"
        orderTablet="0"
        orderWide="0"
        rightOffsetWide="1"
      >
        <div
          className="mfui-beta-store-banner__content"
        >
          <Header
            as="h2"
            className="mfui-beta-store-banner__title"
            color="inherit"
            hAlign="inherit"
          >
            Title
          </Header>
          <Paragraph
            className="mfui-beta-store-banner__text"
            color="inherit"
            hasMargin={false}
          >
            Text
          </Paragraph>
          <div
            className="mfui-beta-store-banner__links"
          >
            <div
              className="mfui-beta-store-banner__stores"
            >
              <Link
                className="mfui-beta-store-banner__store-link mfui-beta-store-banner__store-link_app-store"
                href="App Store link"
              >
                <img
                  alt="Ссылка на приложение в App Store"
                  src="test-file-stub"
                />
              </Link>
              <Link
                className="mfui-beta-store-banner__store-link"
                href="Google Play link"
              >
                <img
                  alt="Ссылка на приложение в Google Play"
                  src="test-file-stub"
                />
              </Link>
            </div>
          </div>
        </div>
      </GridColumn>
      <GridColumn
        all="4"
        desktop="6"
        flex={false}
        grow={false}
        mobile="12"
        orderAll="0"
        orderDesktop="0"
        orderMobile="0"
        orderTablet="0"
        orderWide="0"
        tablet="6"
      >
        <div
          className="mfui-beta-store-banner__device-wrapper"
        >
          <div
            className="mfui-beta-store-banner__device"
          >
            <img
              alt="Изображение приложения на экране телефона"
              className="mfui-beta-store-banner__screen"
              src="image.png"
            />
          </div>
        </div>
      </GridColumn>
    </Grid>
  </div>
</div>
`;
